includes/Media/Video/Trim.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * Class Trim
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Media\Video;
+
+use Google\Web_Stories\Service_Base;
+
+/**
+ * Class Optimization
+ *
+ * @package Google\Web_Stories\Media\Video
+ */
+class Trim extends Service_Base {
+
+	/**
+	 * The trim video post meta key.
+	 *
+	 * @var string
+	 */
+	const TRIM_POST_META_KEY = 'web_stories_trim';
+
+	/**
+	 * Is trim.
+	 *
+	 * @var string
+	 */
+	const IS_TRIM_KEY = 'trim';
+
+	/**
+	 * Register.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @return void
+	 */
+	public function register() {
+		$this->register_meta();
+
+		add_filter( 'wp_prepare_attachment_for_js', [ $this, 'wp_prepare_attachment_for_js' ] );
+	}
+
+	/**
+	 * Register meta for attachment post type.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @return void
+	 */
+	protected function register_meta() {
```

Comment:
For video muting we also used `register_rest_field()`. Why not here? Seems like we should use the same logic for both as they're quite similar.

Reply:
I used it to use register_rest_field, as we needed to support boolean and null.
https://github.com/google/web-stories-wp/blob/0f61a57fac4aa4dd4b56cb4575c55fdcfa5c657b/includes/Media/Video/Muting.php#L123

We don't need this here.